shared/src/main/scala/mlscript/JSBackend.scala

```diff
@@ -344,11 +359,19 @@ class JSBackend {
       case _ => ()
     }
 
+    // Collect names in definitions.
+    otherStmts foreach {
+      case Def(rec, nme, rhs) => defNames += nme
+      case _                  => ()
+    }
+
+    withConstructFnName = getTemporaryName(withConstructFnName)
+
     val defResultObjName = getTemporaryName("defs")
     val exprResultObjName = getTemporaryName("exprs")
     // This hash map counts how many times a name has been used.
     val resolveShadowName = new ShadowNameResolver
-    val stmts: Ls[JSStmt] =
+    var stmts: Ls[JSStmt] =
```

Comment:
Let's not use a needless `var` here. Just add the required statement at the end, like a civilized person 😛

Reply:
This `var` could be avoided for sure. :)